lib/sql_parser.mly

```diff
@@ -362,6 +362,7 @@ expr:
     | e1=expr mnot(IN) l=sequence(expr) { poly Bool (e1::l) }
     | e1=expr mnot(IN) LPAREN select=select_stmt RPAREN { poly Bool [e1; Select (select, `AsValue)] }
     | e1=expr IN table=table_name { Tables.check table; e1 }
+    | e1=expr mnot(IN) p=PARAM { Fun (InF (Var 0, Var 1), [ e1; Inparam (new_param p Any) ]) }
```

Comment:
core of my feedback to this PR is if this can be replaced by smth like `Fun (F (Bool, [Var 0, Var 1]), [e1; Param (new_param p (Tuple Any)])` and I suspect not immediately possible

Reply:
That was my first attempt (which failed), but maybe I can extend the type variable language to be able to represent the polymorphic type `Tuple (Var 0)`, thus getting rid `InF` case. The direct encoding would require using a polymorphic variant with open recursion.